### PR TITLE
Added analytics service account support

### PR DESCRIPTION
**Pull request by rockspore**
Related to https://github.com/apigee/apigee-remote-service-envoy/issues/42

Now the provision command optionally takes a `--service-account` flag (specific to udca service account with role Apigee Analytics Agent) which is the path to the service account json file.

A new k8s Secret will be created with this service account credentials encoded. It will enable the adapter to directly upload analytics to UAP instead of the fluentd endpoint.

The dependency may need to be updated for the GitHub action to succeed. The -golib and then -envoy PRs will need to be merged first.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,9 +2,9 @@ module github.com/apigee/apigee-remote-service-cli
 
 go 1.15
 
-// replace github.com/apigee/apigee-remote-service-golib => ../apigee-remote-service-golib
+//replace github.com/apigee/apigee-remote-service-golib => ../apigee-remote-service-golib
 
-// replace github.com/apigee/apigee-remote-service-envoy => ../apigee-remote-service-envoy
+//replace github.com/apigee/apigee-remote-service-envoy => ../apigee-remote-service-envoy
```
Comment (theganyo) [inline go.mod:7]: Remove these replacements.

File: go.mod [hunk 2]
```diff
@@ -14,5 +14,6 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/cobra v1.0.0
 	go.uber.org/multierr v1.5.0
+	gopkg.in/yaml.v2 v2.3.0
 	gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
```
Comment (theganyo) [inline go.mod:18]: Do we need two versions?

Comment (rockspore) [reply go.mod:18]: No we don't. The VS Code extension keeps inserting yaml.v2 for me...

Comment (theganyo) [reply go.mod:18]: Weird. Does `go mod tidy` clean it up?

Comment (rockspore) [reply go.mod:18]: Unfortunately, `go mod tidy` doesn't always clean up well.

File: cmd/provision/provision.go [hunk 3]
```diff
@@ -103,6 +104,9 @@ to your organization and environment.`,
 	c.Flags().StringVarP(&rootArgs.MFAToken, "mfa", "", "",
 		"Apigee multi-factor authorization token (legacy only)")
 
+	c.Flags().StringVarP(&p.serviceAccount, "service-account", "", "",
```
[the remark this answers is not in the document]
Comment (rockspore) [reply cmd/provision/provision.go]: I changed it to `--analytics-sa` such that it's not too long. Let me know if you have something cooler :)

File: go.mod (hunk 2 above)
Comment (rockspore) [reply go.mod]: Goodbye to yaml.v2

**Review by theganyo — APPROVED** [review]

**Pull request merged**